### ISSUE TITLE
Bug: Time units unknown to InfluxDB if GROUP BY set to "auto"

### ISSUE THREAD

**Issue by joaogb** [grafana/grafana#3814]
**Summary**

I need to set up a few dynamical dashboards in Grafana 2.5.0 with variable templating. The data are stored in an InfluxDB 0.9 database.

It all works well if "GROUP BY time" is set statically (e.g.: 10s) but it fails when "GROUP BY time" is set to "auto".

**Detailed description**

If I set the metric "GROUP BY time" to "auto" in Grafana, the editor mode reads:

> SELECT mean("value") AS "value" FROM "position_x"
> WHERE "geo_city" =~ /$City$/ AND "geo_country" =~ /$Country$/ AND $timeFilter GROUP BY time($interval)

Then, when I try to visualize this particular metric, there is a red error sign on the top left corner which reads:

> InfluxDB Error Response: error parsing query: found y, expected ) at line 1, char 145

When I click on it, I see this query in the "Request parameters" section:

> SELECT mean("value") AS "value" FROM "position_x"
> WHERE "geo_city" =~ /London$/ AND "geo_country" =~ /UK$/ AND time > now() - 7d **GROUP BY time(1y)**

However, according to [InfluxDB Docs v0.9](https://influxdb.com/docs/v0.9/query_language/data_exploration.html):

> Valid units for time() are:
> 
> u microseconds
> s seconds
> m minutes
> h hours
> d days
> w weeks

In short, Grafana is sending "GROUP BY time(1y)" to InfluxDB but "y" is not a valid unit as per InfluxDB's documentation, hence the error.

Any help sorting this issue out will be greatly appreciated.

Thanks!

**Comment by torkelo** [comment]
whats your dashboard time range? think it's the same as this issue, https://github.com/grafana/grafana/issues/2874

**Comment by joaogb** [comment]
> whats your dashboard time range? think it's the same as this issue, #2874

@torkelo My dashboard time range is 1 week (see OP). I see that issue #2874 has been closed. Has this been fixed?

About your comment on #2874:

> its a bug that sometimes happen when panel span gets set to null, bug in panel repeat when no min span is set.

IME this is not happening "sometimes" but pretty much "all the time" when using variable templates.

**Comment by torkelo** [comment]
have you set a min span for the repeated panel?

**Comment by joaogb** [comment]
I had not. I have just set the minimum span and the problem is now solved. Thanks for the quick replies!

**Comment by Manjucloudmpower** [comment]
Panel breaks so often. What's the solution for this?
